frontend/src/pages/onboarding.tsx

```diff
@@ -193,14 +159,128 @@ export default function LoginPage({ csrfToken }: { csrfToken: string }) {
     watch("orgInWhichItParticipates"),
     watch("messageToParticipateOfTheOrg"),
     watch("newUsersEmailsToInvite"),
+    watch("confirmThatItParticipatesOfTheOrg"),
+    watch("newOrganizationName"),
+    watch("newOrganizationDescription"),
+    topics,
+    locations,
+    organizations,
     stepNumber,
   ]);
 
-  const nextStep = () => {
-    if (stepNumber === steps.length - 1) return router.push("/dashboard/newsfeed");
+  const submitOrganizationParticipation = async (selectedValues: any) => {
+
+    const isSelected =  selectedValues.confirmThatItParticipatesOfTheOrg;
+    if (newOrgRequest) {
+      const org_name = selectedValues.newOrganizationName;
+      const org_description = selectedValues.newOrganizationDescription;
+      const dataset_description = selectedValues.newOrganizationDataDescription;
+      if (isSelected && org_name && org_description && apiKey) {
+        const response = await requestNewOrganization({
+          org_name: org_name,
+          org_description: org_description,
+          dataset_description: dataset_description,
+          apiKey: apiKey
+        });
+        if (!response.success) {
+          setErrorMessage("An Error Occurred while Submitting New Organization Request");
+        }
+      }
+    } else {
+      const org_id = selectedValues.orgInWhichItParticipates?.id;
+      const message = selectedValues.messageToParticipateOfTheOrg;
+      if (isSelected && org_id && message && apiKey) {
+        const response = await requestOrganizationOwner({
+          id: org_id,
+          message: message,
+          apiKey: apiKey,
+        });
+        if (!response.success) {
+          setErrorMessage(`An Error Occurred while Submitting Organization ${selectedValues.orgInWhichItParticipates.name} Participation Request.`);
+        }
+      }
+    }
+  };
+  
+  const submitUserInvites = async (selectedValues: any) => {
+    if (selectedValues.newUsersEmailsToInvite && selectedValues.messageToInviteNewUsers) {
+      const response = await inviteUser({
+        emails: selectedValues.newUsersEmailsToInvite,
+        message: selectedValues.messageToInviteNewUsers,
+        apiKey: apiKey,
+      });
+      if (!response.success) {
+        setErrorMessage("An Error Occurred while Sending Invites");
+      }
+    }
+  };
+  
+  const submitFollowPreferences = async () => {
+    const group_ids: string[] = []
+    //add selected topics, locations and organization
+    topics.forEach(topic => {
+      if (topic.selected) {
+        group_ids.push(topic.id);
+      }
+    });
+    locations.forEach(loc => {
+      if (loc.selected) {
+        group_ids.push(loc.id);
+      }
+    });
+    organizations.forEach(org => {
+      if (org.selected) {
+        group_ids.push(org.id);
+      }
+    });
+  
+    if (apiKey && group_ids.length > 0) {
+      const response = await followGroups({
```

Comment:
I suggest using `useQuery` instead of directly calling the `followGroups` function.

Reply:
`useQuery` is mostly used with fetch requests. I guess here I can use `useMutation`.

Reply:
Correct